akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpCookie.java

```diff
@@ -91,6 +118,11 @@ public static HttpCookie create(
      */
     public abstract HttpCookie withHttpOnly(boolean httpOnly);
 
+    /**
+     * Returns a copy of this HttpCookie instance with the given {@link SameSite} set.
+     */
+    public abstract HttpCookie withSameSite(SameSite sameSite);
```

Comment:
What about if you want to unset the attribute?

We also need a getter that converts back to `Optional`.

Reply:
This is now overloaded to also accept an `Optional<SameSite>`.